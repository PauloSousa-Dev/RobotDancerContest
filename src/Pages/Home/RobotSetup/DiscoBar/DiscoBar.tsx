import { useState } from "react";
import { Team } from "../RobotSetup";
import { DanceFloor } from "./DanceFloor";
import { Leaderboard } from "./Leaderboard";

const DiscoBar = ({ getTeams }: { getTeams: Team[] }) => {
  const [renderBoard, setRenderBoard] = useState<boolean>(false);
  return (
    <>
      <h2>Disco Bar</h2>
      {renderBoard && <Leaderboard />}
      <DanceFloor getTeams={getTeams} setRenderBoard={setRenderBoard} />
    </>
  );
};

export { DiscoBar };
