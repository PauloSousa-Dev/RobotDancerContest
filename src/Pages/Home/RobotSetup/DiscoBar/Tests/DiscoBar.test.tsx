import { render } from "@testing-library/react";
import { describe, expect, test } from "@jest/globals";
import { DiscoBar } from "../DiscoBar";

jest.mock("@/Contexts");

describe("Robot Setup component", () => {
  test("should render correctly", () => {
    const { container } = render(<DiscoBar />);
    expect(container).toMatchSnapshot();
  });
});
