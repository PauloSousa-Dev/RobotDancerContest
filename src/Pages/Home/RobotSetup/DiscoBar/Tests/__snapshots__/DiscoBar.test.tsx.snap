// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Robot Setup component should render correctly 1`] = `
<div>
  <h2>
    Disco Bar
  </h2>
  <div
    class="wrapper"
  >
    <div
      class="teamWrapper"
    >
      <div>
        <span
          class="teamName"
        >
          Team 
          Team 1
        </span>
      </div>
      <div>
        <span
          class="teamName"
        >
          Team 
          Team 2
        </span>
      </div>
    </div>
    <div
      class="buttonWrapper"
    >
      <button
        class="btn btn-primary btn-medium"
        type="button"
      >
        Start Contest
      </button>
    </div>
  </div>
</div>
`;
